bridge-adapters/kinetic-bridgehub-adapter-cherwell/src/main/java/com/kineticdata/bridgehub/adapter/cherwell/CherwellAdapter.java

```diff
@@ -0,0 +1,758 @@
+    
+    /**
+     * Ensure that the sort order list is linked so that order can not be changed.
+     * 
+     * @param uncastSortOrderItems
+     * @return
+     * @throws IllegalArgumentException 
+     */
+    private LinkedHashMap<String, String> 
+        getSortOrderItems (Map<String, String> uncastSortOrderItems)
+        throws IllegalArgumentException{
+        
+        /* results of parseOrder does not allow for a structure that 
+         * guarantees order.  Casting is required to preserver order.
+         */
+        if (!(uncastSortOrderItems instanceof LinkedHashMap)) {
+            throw new IllegalArgumentException("Sort Order Items was invalid.");
+        }
+        
+        return (LinkedHashMap)uncastSortOrderItems;
+    }
+    
+    /**************************** Path Definitions ****************************/
+    
+    /**
+     * 
+     * Default requires single path and retrieve parameters must contain an "id"
+     * @param structureList
+     * @param parameters
+     * @return
+     * @throws BridgeError 
+     * 
+     */
+    protected static String pathTeams(List<String> structureList,
+        Map<String, String> parameters) throws BridgeError {
+        
+        String path = PATH;
+        
+        if (parameters.containsKey("id")) {
+            path += "/V1/getteam/" + parameters.get("id");
+            parameters.remove("id");
+        } else {
+            path += "/V1/getteams";
+        }
+        
+        return path;
+    }
+
+    /**
+     * Build a path to get saved search results for v1 and v2 along with running saved search by name and internal id.
+     *
+     * @param structureList
+     * @param parameters
+     * @return
+     * @throws BridgeError
+     */
+    protected static String pathSavedSearch(List<String> structureList, Map<String, String> parameters) throws BridgeError {
+        String path = PATH;
+
+        switch (structureList.get(1)) {
+            case "Internal ID":
+            case "Name":
+
+                // Search by Internal Id or Name
+                String searchType = "";
+                if (structureList.get(1).equals("Internal ID")) {
+                    searchType = "searchid";
+                } else {
+                    searchType = "searchName";
+                }
+
+                if (!(parameters.containsKey("association")
+                        && parameters.containsKey("scope")
+                        && parameters.containsKey("scopeowner")
+                        && parameters.containsKey(searchType))) {
+                    throw new BridgeError("The Saved Search > "+ structureList.get(1) +" structure requires \"association\", \"scope\"," +
+                            " \"scopeowner\", and \""+ searchType +"\"");
+                }
+
+                String searchTypeValue;
+                try {
+                    searchTypeValue = URLEncoder.encode(parameters.get(searchType), StandardCharsets.UTF_8.toString()).replace("+", "%20");
```

Comment:
Probably create a static method to handle the encoding as it seems it should be used in many places.

```
static String urlEncode(String input) {
    return URLEncoder
        .encode(input, StandardCharsets.UTF_8.toString())
        .replace("+", "%20");
}
```

Then `searchTypeValue = urlEncode(parameters.get(searchType));`